src/vm/hints/uint256_utils.rs

```diff
@@ -513,4 +672,663 @@ mod tests {
+                    ap_tracking_data: None,
+                },
+            ),
+            (
+                1,
+                HintReference {
+                    register: Register::FP,
+                    offset1: 0,
+                    offset2: 0,
+                    inner_dereference: false,
+                    ap_tracking_data: None,
+                },
+            ),
+        ]);
+
+        //Insert  ids.n.low into memory
+        vm.memory
+            .insert(
+                &MaybeRelocatable::from((1, 0)),
+                &MaybeRelocatable::from(bigint!(17)),
+            )
+            .unwrap();
+
+        //Insert ids.n.high into memory
+        vm.memory
+            .insert(
+                &MaybeRelocatable::from((1, 1)),
+                &MaybeRelocatable::from(bigint!(7)),
+            )
+            .unwrap();
+
+        //Insert a value in the ids.root.low address so the hint insert fails
+        vm.memory
+            .insert(
+                &MaybeRelocatable::from((1, 5)),
+                &MaybeRelocatable::from(bigint!(1)),
+            )
+            .unwrap();
+
+        //Execute the hint
+        assert_eq!(
+            execute_hint(&mut vm, hint_code, ids, &ApTracking::new()),
+            Err(VirtualMachineError::MemoryError(
+                MemoryError::InconsistentMemory(
+                    MaybeRelocatable::from((1, 5)),
+                    MaybeRelocatable::from(bigint!(1)),
+                    MaybeRelocatable::from(bigint_str!(b"48805497317890012913")),
+                )
+            ))
+        );
+    }
+
+    #[test]
+    fn run_signed_nn_ok_result_one() {
+        let hint_code = "memory[ap] = 1 if 0 <= (ids.a.high % PRIME) < 2 ** 127 else 0".as_bytes();
+        let mut vm = VirtualMachine::new(
+            BigInt::new(Sign::Plus, vec![1, 0, 0, 0, 0, 0, 17, 134217728]),
+            vec![(
+                "range_check".to_string(),
+                Box::new(RangeCheckBuiltinRunner::new(true, bigint!(8), 8)),
+            )],
+            false,
+        );
+        for _ in 0..3 {
+            vm.segments.add(&mut vm.memory, None);
+        }
+
+        //Initialize fp
+        vm.run_context.fp = MaybeRelocatable::from((1, 4));
+        vm.run_context.ap = MaybeRelocatable::from((1, 5));
+
+        //Create ids
+        let mut ids = HashMap::<String, BigInt>::new();
+        ids.insert(String::from("a"), bigint!(0));
+
+        //Create references
+        vm.references = HashMap::from([(
+            0,
+            HintReference {
+                register: Register::FP,
+                offset1: -4,
+                offset2: 0,
+                inner_dereference: false,
+                ap_tracking_data: None,
+            },
+        )]);
+
+        //Insert ids.a.high into memory
+        vm.memory
+            .insert(
+                &MaybeRelocatable::from((1, 1)),
+                &MaybeRelocatable::from(bigint!(2).pow(127) - 1 + &vm.prime),
```

Comment:
```suggestion
                &MaybeRelocatable::from(&vm.prime + i128::MAX),
```
I _think_ that should work directly. Otherwise, you can still construct the `BigInt` with the macro and `i128::MAX`.

Reply:
`&vm.prime + i128::MAX` works Ok